common/interactive-object/server.mjs

```diff
@@ -259,6 +262,52 @@ class InteractiveObjectServerBaseClass {
 
     return this.sf_exit(playerID, {next: this.dataStore.get(playerID)});
   }
+
+  /**
+   * Check permission by JWT token and go to corresponding state
+   * @param {String} playerID
+   * @param {Object} kwargs - TODO
+   * @return {String} - the next state
+   */
+  async sf_checkPermission(playerID, kwargs) {
+    const permission = await this.helper.getToken(playerID);
+    permission.scope.push('online_bulletin_editor');
+    const {options} = kwargs;
+    for (const [identity, nextState] of Object.entries(options)) {
+      if (permission.scope.includes(identity)) {
+        return nextState;
+      }
+    }
+    // No identity match
+    throw 'No identity match';
```

Comment:
We might need another kwargs parameter for this, so that the user can catch the conditions whereby no scope is matched.

Reply:
Now the exception is catched [here](https://github.com/canyugs/hitcon-online/blob/main/common/interactive-object/server.mjs#L142)